client/client.go

```diff
@@ -150,6 +152,37 @@ func FromEnv(c *Client) error {
 	return nil
 }
 
+// WithContextStoreOrEnv applies a context store config, or classic docker env if detected
+// Heuristic to use either context store or legacy DOCKER_* variables is:
+// if contextNameOverride is set -> load this context from context store
+// if DOCKER_CONTEXT env variable is set -> load this context from context store
+// if DOCKER_HOST is set -> fallback to FromEnv
+// if a current context is set in the store -> load this context
```

Comment:
Do you have a TODO to cause the documentation to be updated? (Seems to not be in this repo)

Reply:
Nice catch. @thaJeztah where should we put this kind of documentation (e.g.: now the preferred way to initialize a client is to use `WithContextStoreOrEnv`)?